src/main/java/de/zalando/aruha/nakadi/enrichment/Enrichment.java

```diff
@@ -0,0 +1,49 @@
+package de.zalando.aruha.nakadi.enrichment;
+
+import de.zalando.aruha.nakadi.domain.EnrichmentStrategyDescriptor;
+import de.zalando.aruha.nakadi.domain.EventCategory;
+import de.zalando.aruha.nakadi.domain.EventType;
+import de.zalando.aruha.nakadi.exceptions.EnrichmentException;
+import de.zalando.aruha.nakadi.exceptions.InvalidEventTypeException;
+import org.json.JSONObject;
+
+import java.util.HashSet;
+import java.util.Set;
+
+public class Enrichment {
+    private final EnrichmentsRegistry registry;
+
+    public Enrichment(final EnrichmentsRegistry registry) {
+        this.registry = registry;
+    }
+
+    public void validate(final EventType eventType) throws InvalidEventTypeException {
```

Comment:
Since I'm using an `enum` for `EnrichmentStrategyDescriptor`, if the user provides an invalid value, then he will see a 400 answer.

```
{"type":"http://httpstatus.es/400","title":"Bad Request","status":400,"detail":"Illegal enum value: 'metadata_strategy'. Possible values: [metadata_enrichment] (through reference chain: de.zalando.aruha.nakadi.domain.EventType[\"enrichment_strategies\"]->java.util.ArrayList[0])"}
```

I know it's ugly to expose internals, but that's how spring does with it's validations.